tools/c7n_mailer/c7n_mailer/email_delivery.py

```diff
@@ -211,43 +211,43 @@ def get_email_to_addrs_to_resources_map(self, sqs_message):
             resource_emails = tuple(sorted(set(resource_emails)))
             # only if there are valid emails available, add it to the map
             if resource_emails:
-                email_to_addrs_to_resources_map.setdefault(resource_emails, []).append(resource)
-        if email_to_addrs_to_resources_map == {}:
+                emails_to_resources_map.setdefault(resource_emails, []).append(resource)
+        if emails_to_resources_map == {}:
             self.logger.debug("Found no email addresses, sending no emails.")
         # eg: { ('milton@initech.com', 'peter@initech.com'): [resource1, resource2, etc] }
-        return email_to_addrs_to_resources_map
+        return emails_to_resources_map
 
-    def get_to_addrs_email_messages_map(self, sqs_message):
-        to_addrs_to_resources_map = self.get_email_to_addrs_to_resources_map(sqs_message)
-        to_addrs_to_mimetext_map = {}
-        for to_addrs, resources in to_addrs_to_resources_map.items():
-            to_addrs_to_mimetext_map[to_addrs] = get_mimetext_message(
+    def get_emails_to_mimetext_map(self, sqs_message):
+        emails_to_resources_map = self.get_emails_to_resources_map(sqs_message)
+        emails_to_mimetext_map = {}
+        for to_addrs, resources in emails_to_resources_map.items():
+            emails_to_mimetext_map[to_addrs] = get_mimetext_message(
                 self.config, self.logger, sqs_message, resources, list(to_addrs)
             )
         # eg: { ('milton@initech.com', 'peter@initech.com'): mimetext_message }
-        return to_addrs_to_mimetext_map
+        return emails_to_mimetext_map
 
-    def send_c7n_email(self, sqs_message, email_to_addrs, mimetext_msg):
+    def send_c7n_email(self, sqs_message):
+        emails_to_mimetext_map = self.get_emails_to_mimetext_map(sqs_message)
+        email_to_addrs = list(emails_to_mimetext_map.keys())
```

Comment:
This tripped me up at first glance because we're defining `email_to_addrs` here, then redefining it in 2/3 of the code paths below, then referencing it in log calls after it may or may not have been redefined.

I _think_ it would be clearer to use one variable here and in the `self.logger.*` calls below, and a separate one inside the smtp loop.

To be clear I tested this PR locally and it seemed to work fine, but the logic wasn't immediately obvious to me because of how `email_to_addrs` was only sometimes redefined.

Reply:
I agree that's rather odd because I was struggling if we should keeping the original `self.logger.error` email_to_addrs scope, which print partial of the "to" only.

I updated the code, and hope I understand your comment correctly. Plaese review again.